# SIPTA -- Ingesta y EDA: Educación y Cobertura Escolar
**Proyecto**: Sistema de Indicadores y Priorización Territorial y Alertas Tempranas (DataJam Bogotá)  
**Fase PDCO**: DEVELOPMENT | **Fase CRISP-DM**: Data Understanding / Data Preparation  
**Autoría**: **Persona A (Adan Sánchez -- Análisis Exploratorio de Datos / EDA)**  
**Relevamiento de fuentes**: Persona B (Yesid Bello -- Data Scientist)  
**Objetivo**: Ingesta de oferta de cupos, colegios SED, matrícula oficial y análisis exploratorio educativo (EDA).  
**Datos de Entrada**: `data/raw/EDUCACION/*`  
**Datos de Salida**: `data/processed/EDUCACION/*`


## 1. Ingesta y Carga de Sedes, Matrícula y Oferta de Cupos



# SIPTA Notebook: Ingestión de datos -- Educación

## Dominio: Educación

Este notebook documenta la ingesta reproducible del dataset oficial
**Oferta de cupos del sector oficial en Bogotá D.C.**

### Objetivos

- Cargar el archivo original almacenado en `data/raw/EDUCACION`.
- Preservar el archivo crudo sin modificaciones.
- Verificar el formato geoespacial y su sistema de referencia.
- Registrar dimensiones, columnas, tipos de datos y geometría.
- Verificar que el dataset queda disponible para la fase posterior de validación.

> En este notebook no se realizan limpieza, corrección de registros, integración territorial ni cálculo de indicadores.

In [3]:
from pathlib import Path
import geopandas as gpd
import pandas as pd

PROJECT_ROOT = None
for _p in [Path('.').resolve(), Path('.').resolve().parent, Path('.').resolve().parent.parent]:
    if (_p / 'data' / 'raw').exists() or (_p / 'src').exists():
        PROJECT_ROOT = _p
        break
if PROJECT_ROOT is None:
    PROJECT_ROOT = Path('.').resolve()

RAW_DIR = PROJECT_ROOT / "data" / "raw"
EDUCACION_DIR = RAW_DIR / "EDUCACION"

archivo_educacion = EDUCACION_DIR / "ofertacupos_032025.geojson"

print("PROJECT ROOT:", PROJECT_ROOT)
print("Archivo Educación:", archivo_educacion)
print("¿Existe?:", archivo_educacion.exists())


PROJECT ROOT: /content
Archivo Educación: /content/data/raw/EDUCACION/ofertacupos_032025.geojson
¿Existe?: False


In [4]:
print("=== CARGAR DATASET CRUDO ===")

gdf_educacion = gpd.read_file(archivo_educacion)

print("Carga completada.")
print("Tipo:", type(gdf_educacion).__name__)
print("Filas:", gdf_educacion.shape[0])
print("Columnas:", gdf_educacion.shape[1])

gdf_educacion.head()



=== CARGAR DATASET CRUDO ===


DataSourceError: /content/data/raw/EDUCACION/ofertacupos_032025.geojson: No such file or directory

In [ ]:
print("=== COLUMNAS ===")

for i, columna in enumerate(gdf_educacion.columns, start=1):
    print(f"{i}. {columna}")


=== COLUMNAS ===
1. NOMBRE_EST
2. GENERO
3. COD_LOCA
4. CLASE_TIPO
5. FECHA
6. OPreescola
7. OPrimaria
8. OSecundari
9. OMedia
10. OTotal
11. Aceleracio
12. DANE12_EST
13. Educacion_
14. geometry


In [ ]:
print("=== TIPOS DE DATOS ORIGINALES ===")
print(gdf_educacion.dtypes)


=== TIPOS DE DATOS ORIGINALES ===
NOMBRE_EST               str
GENERO                 int32
COD_LOCA                 str
CLASE_TIPO             int32
FECHA         datetime64[ms]
OPreescola             int32
OPrimaria              int32
OSecundari             int32
OMedia                 int32
OTotal                 int32
Aceleracio             int32
DANE12_EST               str
Educacion_             int32
geometry            geometry
dtype: object


In [ ]:
print("=== PRIMERAS 5 FILAS ===")
display(gdf_educacion.head())

print("\n=== ÚLTIMAS 5 FILAS ===")
display(gdf_educacion.tail())


=== PRIMERAS 5 FILAS ===


,NOMBRE_EST,GENERO,COD_LOCA,CLASE_TIPO,FECHA,OPreescola,OPrimaria,OSecundari,OMedia,OTotal,Aceleracio,DANE12_EST,Educacion_,geometry
0,COLEGIO AQUILEO PARRA (IED),5,01,1,2025-03-31,170,1115,1045,313,2920,0,111001000132,277,POINT (-8242002.703 531131.432)
1,COLEGIO AGUSTIN FERNANDEZ (IED),5,01,1,2025-03-31,0,317,599,262,1500,0,111001029955,322,POINT (-8240325.118 527101.196)
2,COLEGIO AGUSTIN FERNANDEZ (IED),5,01,1,2025-03-31,40,200,0,0,240,0,111001029955,0,POINT (-8239609.334 527422.892)
3,COLEGIO AGUSTIN FERNANDEZ (IED),5,01,1,2025-03-31,75,350,0,0,425,0,111001029955,0,POINT (-8239442.355 527705.494)
4,COLEGIO AGUSTIN FERNANDEZ (IED),5,01,1,2025-03-31,58,70,0,0,202,74,111001029955,0,POINT (-8240365.193 527535.709)



=== ÚLTIMAS 5 FILAS ===


,NOMBRE_EST,GENERO,COD_LOCA,CLASE_TIPO,FECHA,OPreescola,OPrimaria,OSecundari,OMedia,OTotal,Aceleracio,DANE12_EST,Educacion_,geometry
742,COLEGIO GIMNASIO DEL CAMPO JUAN DE LA CRUZ VAR...,5,20,1,2025-03-31,12,29,0,0,41,0,211850001473,0,POINT (-8272596.639 448876.315)
743,COLEGIO GIMNASIO DEL CAMPO JUAN DE LA CRUZ VAR...,5,20,1,2025-03-31,3,6,0,0,9,0,211850001473,0,POINT (-8273964.755 451988.761)
744,COLEGIO GIMNASIO DEL CAMPO JUAN DE LA CRUZ VAR...,5,20,1,2025-03-31,4,9,0,0,13,0,211850001473,0,POINT (-8274258.639 456150.389)
745,COLEGIO GIMNASIO DEL CAMPO JUAN DE LA CRUZ VAR...,5,20,1,2025-03-31,4,7,0,0,11,0,211850001473,0,POINT (-8275905.054 446613.191)
746,COLEGIO GIMNASIO DEL CAMPO JUAN DE LA CRUZ VAR...,5,20,1,2025-03-31,2,8,0,0,10,0,211850001473,0,POINT (-8276697.649 449169.812)


In [ ]:
print("=== INFORMACIÓN GEOESPACIAL ===")

print("CRS:", gdf_educacion.crs)

print("\nTipos de geometría:")
print(gdf_educacion.geometry.geom_type.value_counts())

print("\nGeometrías nulas:")
print(gdf_educacion.geometry.isna().sum())

print("\nGeometrías vacías:")
print(gdf_educacion.geometry.is_empty.sum())


=== INFORMACIÓN GEOESPACIAL ===
CRS: EPSG:3857

Tipos de geometría:
Point    747
Name: count, dtype: int64

Geometrías nulas:
0

Geometrías vacías:
0


## Fuente y trazabilidad

**Dataset:** Oferta de cupos sector oficial Bogotá D.C.  
**Fuente:** Portal de Datos Abiertos de Bogotá  
**Formato original:** GeoJSON  
**Archivo local:** `data/raw/EDUCACION/ofertacupos_032025.geojson`

### Características técnicas observadas

- Registros cargados: **747**.
- Variables: **14**.
- Tipo de estructura: `GeoDataFrame`.
- Sistema de referencia espacial: **EPSG:3857**.
- Tipo de geometría: **Point**.
- Geometrías nulas: **0**.
- Geometrías vacías: **0**.
- El archivo se conserva en `data/raw/EDUCACION` sin modificaciones manuales.

### Alcance de esta etapa

La ingesta únicamente comprueba que el dataset original puede cargarse de forma
reproducible y que conserva su estructura geoespacial.

La evaluación de valores nulos, duplicados, coherencia de cupos, identificadores
DANE, códigos de localidad y consistencia territorial se realiza en
`02_validation_educacion.ipynb`.

En esta etapa no se realizan correcciones, limpieza, integración territorial ni
cálculo de indicadores.

## 2. Análisis Exploratorio de Datos (EDA) del Sector Educación



# 03 · EDA Educación

Dos fuentes de la Secretaría de Educación (SED) en `data/raw/EDUCACION/`:
- `colegios122025.gpkg` -- sedes educativas (capa `colegios_122025`, ~2.211 puntos, EPSG:3857).
- `matricula_total_colegios_oficiales.gpkg` -- matrícula total por sede oficial (capa
  `matriculatotal_042025`, 747 filas, 55 columnas, EPSG:3857).

**Indicadores objetivo**: EDU-01 (cobertura educativa por localidad), EDU-02 (matrícula oficial
por localidad). Ambas usan `COD_LOCA` (código 1-20) como clave territorial.

## 0. Configuración

In [ ]:
import sys, os, re, json, warnings, time
from pathlib import Path
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import geopandas as gpd
warnings.filterwarnings("ignore")

# Búsqueda universal de la raíz del proyecto (Local, Subdirectorios y Google Colab)
ROOT = None
for _p in [Path(".").resolve(), Path("..").resolve(), Path("../..").resolve(), Path("/content/DataJam_DataOlinguitos_Gen"), Path("/content")]:
    if (_p / "src" / "eda").exists() or (_p / "src").exists():
        ROOT = _p
        break

if ROOT is None:
    ROOT = Path(".").resolve()

if str(ROOT) not in sys.path:
    sys.path.insert(0, str(ROOT))

RAW_DIR = ROOT / "data" / "raw"
MR_PATH = RAW_DIR / "INFRAESTRUCTURA_ESPACIO_PUBLICO" / "gpkg_mr_v03.26" / "gpkg_mr_v03.26.gpkg"
REPORTS = ROOT / "reports" / "eda"
PERFILES = REPORTS / "perfiles"
CACHE = REPORTS / "cache"
TIEMPOS = REPORTS / "tiempos"
for _d in (REPORTS, PERFILES, CACHE, TIEMPOS):
    _d.mkdir(parents=True, exist_ok=True)

SMOKE = os.environ.get("EDA_SMOKE") == "1"
print("ROOT:", ROOT, "| SMOKE:", SMOKE)

import src.eda as eda
from src.eda.explore import explorar_dataset

_SECTION_T = {}

def t0(nombre):
    _SECTION_T[nombre] = time.time()

def t1(nombre):
    _SECTION_T[nombre] = time.time() - _SECTION_T[nombre]

def guardar_tiempos(archivo):
    df = pd.DataFrame([{"seccion": k, "segundos": round(v, 1)} for k, v in _SECTION_T.items()])
    df.to_csv(TIEMPOS / archivo, index=False)
    print(f"tiempos guardados: {archivo} ({len(df)} secciones)")


## 0.1 Fuentes del sector en el catálogo

In [ ]:
cat = eda.load_catalog()
sec = cat[cat["indicadores"].astype(str).str.contains("EDU", case=False, na=False)]
display(sec[["id", "nombre", "archivo", "temporalidad", "indicadores", "valor_publico"]])


## 1. Sedes educativas

### Sedes educativas -- SED

In [ ]:
t0('colegios')
SPEC = {
    'id': 'colegios_122025',
    'titulo': 'Sedes educativas -- SED',
    'path': 'data/raw/EDUCACION/colegios122025.gpkg',
    'capa': 'colegios_122025',
    'origen': 'Secretaría de Educación del Distrito (SED)',
    'corte': 'Dic 2025 (nombre del archivo)',
    'valor_publico': 'Oferta educativa por localidad: sedes, sector, calendario, carácter, estrato',
    'indicadores': 'EDU-01 (sedes por localidad)',
    'notas': 'COD_LOCA es código numérico 1-20: se mapea a nombre con localidad_de_codigo. 30 columnas.',
}
RES = explorar_dataset(SPEC, RAW_DIR, PERFILES, smoke=SMOKE)
t1('colegios')


## 2. Matrícula total oficial

### Matrícula total por sede oficial -- SED

In [ ]:
t0('matricula')
SPEC = {
    'id': 'matricula_oficial_042025',
    'titulo': 'Matrícula total por sede oficial -- SED',
    'path': 'data/raw/EDUCACION/matricula_total_colegios_oficiales.gpkg',
    'capa': 'matriculatotal_042025',
    'origen': 'Secretaría de Educación del Distrito (SED)',
    'corte': 'Abr 2025 (nombre del archivo)',
    'valor_publico': 'Matrícula oficial por localidad, con desagregación por discapacidad, etnia y talentos',
    'indicadores': 'EDU-02 (matrícula por localidad)',
    'notas': '55 columnas: TMATRIC_GE es la matrícula general; columnas DA_*/DI_*/DV_*/DIS_*/T_* agregan condiciones de discapacidad, etnia y talentos. COD_LOCA numérico 1-20.',
}
RES = explorar_dataset(SPEC, RAW_DIR, PERFILES, smoke=SMOKE)
t1('matricula')


## 3. Análisis de cobertura educativa

### 3.1 Sedes por localidad y mapa

In [ ]:
from src.eda.profiling import localidad_de_codigo
from src.eda.spatial import load_loca
col = gpd.read_file(str(RAW_DIR / "EDUCACION" / "colegios122025.gpkg"), layer="colegios_122025")
col["LOCALIDAD"] = col["COD_LOCA"].map(localidad_de_codigo)
conteo = col.groupby("LOCALIDAD").size().sort_values(ascending=False)
conteo.to_csv(REPORTS / "educacion_colegios_por_localidad.csv", index=False)
display(conteo.rename("sedes"))
print("Total sedes:", len(col), "| localidades con sedes:", conteo.notna().sum())
conteo.plot(kind="bar", figsize=(11, 4), title="Sedes educativas por localidad", color="#4c72b0")
plt.ylabel("sedes")
plt.show()

fig, ax = plt.subplots(figsize=(9, 9))
loca = load_loca(MR_PATH)
loca.plot(ax=ax, color="#f0f0e8", edgecolor="#9a9a8a", linewidth=0.4)
col.to_crs(loca.crs).plot(ax=ax, color="#2ca02c", markersize=3, alpha=0.6)
ax.set_title(f"Sedes educativas ({len(col)})")
ax.set_axis_off()
plt.show()


### 3.2 Matrícula oficial por localidad

In [ ]:
mat = gpd.read_file(str(RAW_DIR / "EDUCACION" / "matricula_total_colegios_oficiales.gpkg"), layer="matriculatotal_042025")
mat["LOCALIDAD"] = mat["COD_LOCA"].map(localidad_de_codigo)
tot = mat.groupby("LOCALIDAD")["TMATRIC_GE"].sum().sort_values(ascending=False)
tot.to_csv(REPORTS / "educacion_matricula_por_localidad.csv", index=False)
print("Colegios con matrícula:", len(mat), "| matrícula total:", f"{mat['TMATRIC_GE'].sum():,.0f}")
tot.plot(kind="bar", figsize=(11, 4), title="Matrícula oficial por localidad", color="#55a868")
plt.ylabel("estudiantes")
plt.show()


### 3.3 Matrícula: discapacidad, etnia y talentos

In [ ]:
disc = [c for c in mat.columns if c.startswith(("DA_", "DI_", "DV_", "DIS_", "T_")) or c in ("SORDOCEGUE", "OTRA")]
et_pct = mat["TOT_EST_ET"].sum() / mat["TMATRIC_GE"].sum()
disc_pct = mat[disc].sum().sum() / mat["TMATRIC_GE"].sum()
print(f"Estudiantes con condición de discapacidad: {disc_pct:.2%}")
print(f"Población étnica (afro, indígena, etc.): {et_pct:.2%}")
tal = mat[["TE_ACTFISI", "TE_ARTES", "TE_CNATUR", "TE_CSOCIAL", "TE_LDSOCIA", "TE_TECNOLO"]].sum()
tal.plot(kind="bar", figsize=(8, 4), title="Estudiantes por talento excepcional", color="#8172b3")
plt.ylabel("estudiantes")
plt.show()


## 4. Indicadores del sector

In [ ]:
sts = eda.indicator_status()
sts_sec = sts[sts["indicador"].str.startswith("EDU", na=False)]
display(sts_sec[["indicador", "dimension", "estado", "que_falta"]])
sts_sec.to_csv(REPORTS / "indicadores_educacion.csv", index=False)
guardar_tiempos("03_eda_educacion.csv")
print("Secciones del notebook:", list(_SECTION_T.keys()))
